# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLExtensionDebugShaders.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLExtensionDebugShaders.cpp
--- a/dom/canvas/WebGLExtensionDebugShaders.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,35 +0,0 @@
-/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "WebGLExtensions.h"
-
-#include "mozilla/dom/WebGLRenderingContextBinding.h"
-#include "WebGLContext.h"
-#include "WebGLShader.h"
-
-namespace mozilla {
-
-WebGLExtensionDebugShaders::WebGLExtensionDebugShaders(WebGLContext* webgl)
-    : WebGLExtensionBase(webgl) {}
-
-WebGLExtensionDebugShaders::~WebGLExtensionDebugShaders() {}
-
-// If no source has been defined, compileShader() has not been called, or the
-// translation has failed for shader, an empty string is returned; otherwise,
-// return the translated source.
-nsString WebGLExtensionDebugShaders::GetTranslatedShaderSource(
-    const WebGLShader& shader) const {
-  if (mIsLost) return {};
-
-  const WebGLContext::FuncScope funcScope(*mContext,
-                                          "getShaderTranslatedSource");
-  MOZ_ASSERT(!mContext->IsContextLost());
-
-  if (!mContext->ValidateObject("shader", shader)) return EmptyString();
-
-  return shader.GetShaderTranslatedSource();
-}
-
-}  // namespace mozilla